src/frontend/web_application/src/components/ImportContactForm/presenter.jsx

```diff
@@ -1,139 +1,188 @@
+
+const File = ({ file, onClick }) => (
+  <div className="m-import-contact-form__file">
+    <span className="m-import-contact-form__file-name">{file.name}</span>
+    <span className="m-import-contact-form__file-size">{file.size / 1000} ko</span>
+    <Button
+      className="m-import-contact-form__remove-button"
+      display="inline"
+      icon="remove"
+      value={file}
+      onClick={onClick}
+    />
+  </div>
+);
+
+File.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  file: PropTypes.shape({}).isRequired,
+};
+
+const InputFile = ({ onChange, errors, __ }) => (
+  <div>
+    <label htmlFor="files[]" className="m-import-contact-form__label">
+      <span className="m-import-contact-form__label-button"><Icon type="plus" /></span>
+      <span className="m-import-contact-form__label-text">{__('import-contact.form.add_a_file.label')}</span>
+      <span className="m-import-contact-form__label-icon"><Icon type="folder" /></span>
+      <input
+        id="files"
+        type="file"
+        name="files[]"
+        className="m-import-contact-form__input"
+        onChange={onChange}
+        accept=".vcf, .vcard"
+      />
+    </label>
+    { errors &&
+      <FieldErrors errors={errors} />
+    }
+  </div>
+);
+
+InputFile.propTypes = {
+  __: PropTypes.func.isRequired,
+  onChange: PropTypes.func.isRequired,
+  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
+};
+
 
 class ImportContactForm extends Component {
   static propTypes = {
-    uploadFile: PropTypes.func,
+    onSubmit: PropTypes.func.isRequired,
     onCancel: PropTypes.func,
+    errors: PropTypes.shape({}),
     __: PropTypes.func.isRequired,
+    hasImported: PropTypes.bool,
   };
 
   static defaultProps = {
-    uploadFile: null,
     onCancel: null,
+    errors: {},
+    hasImported: false,
   }
 
   constructor(props) {
     super(props);
     this.state = {
-      files: [],
-      errors: [],
+      file: null,
+      fieldError: [],
+      formData: {},
     };
 
-    this.onInputChange = this.onInputChange.bind(this);
-    this.onResetForm = this.onResetForm.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmitForm = this.handleSubmitForm.bind(this);
+    this.resetForm = this.resetForm.bind(this);
     this.renderButtons = this.renderButtons.bind(this);
+    this.validateField = this.validateField.bind(this);
   }
 
-  onInputChange(ev) {
-    const files = ev.target.files;
-    if (files.length > 0) {
-      const file = ev.target.files[0];
-      this.validate(file);
-    }
+  handleInputChange(ev) {
+    const files = ev.target.files.length > 0 ? ev.target.files : null;
+    if (files) { this.validateField(files[0]); }
+  }
+
+  handleSubmitForm(ev) {
+    ev.preventDefault();
+    const { formData } = this.state;
+    this.props.onSubmit({ formData });
   }
 
-  onResetForm() {
+  resetForm() {
     document.getElementById('import-contact-form').reset();
```

Comment:
ok! this line was useless: no need to reset form because when state.file is null, it renders a form without any value in `<input>`.